waspc/src/Wasp/AppSpec/Job.hs

```diff
@@ -15,22 +20,57 @@ import Wasp.AppSpec.JSON (JSON (..))
 
 data Job = Job
   { executor :: JobExecutor,
-    perform :: Perform
+    perform :: Perform,
+    schedule :: Maybe Schedule
   }
   deriving (Show, Eq, Data)
 
 instance IsDecl Job
 
-data JobExecutor = Passthrough | PgBoss
+data JobExecutor = Simple | PgBoss
   deriving (Show, Eq, Data, Ord, Enum, Bounded)
 
 data Perform = Perform
   { fn :: ExtImport,
-    options :: Maybe JSON
+    executorOptions :: Maybe ExecutorOptions
   }
   deriving (Show, Eq, Data)
 
 instance IsDecl Perform
 
+-- Allows jobs to run via some cron schedule.
+data Schedule = Schedule
+  { cron :: String, -- 5 field cron expression, exe: "*/5 * * * *".
+    args :: Maybe JSON, -- Arguments to pass to the job handler function (`Perform.fn`).
+    executorOptions :: Maybe ExecutorOptions
+  }
+  deriving (Show, Eq, Data)
+
+instance IsDecl Schedule
+
+-- These are optional executor-specific JSON options we pass
+-- directly through to the executor when submitting jobs.
+data ExecutorOptions = ExecutorOptions
+  { pgBoss :: Maybe JSON,
+    simple :: Maybe JSON
+  }
+  deriving (Show, Eq, Data)
+
 jobExecutors :: [JobExecutor]
 jobExecutors = enumFrom minBound :: [JobExecutor]
+
+-- Helpers to disambiguate duplicate field `executorOptions`.
+performExecutorOptionsJson :: Job -> Maybe JSON
+performExecutorOptionsJson job =
+  executorOptionsJson (executor job) (executorOptions (perform job :: Perform))
+
+scheduleExecutorOptionsJson :: Job -> Maybe JSON
+scheduleExecutorOptionsJson job =
+  case schedule job of
```

Comment:
I think you can do:
```hs
do
  s <- schedule job
  executorOptionsJson (executor job) (executorOptions (s :: Schedule))
```

which I believe can then also be shortened to
```hs
schedule job >>= executorOptionsJson (executor job) . executorOptions
```

Reply:
Ah yes! Didn't at first see I was already in a `Maybe` dealing with `Maybe`s. Thnx! Will update 👍🏻